arelle/plugin/validate/NL/PluginValidationDataExtension.py

```diff
@@ -45,6 +49,26 @@ def contextsByDocument(self, modelXbrl: ModelXbrl) -> dict[str, list[ModelContex
         self._contextsByDocument = dict(contextsByDocument)
         return self._contextsByDocument
 
+
+    def checkContexts(self, allContexts: dict[str, list[ModelContext]]) -> None:
+        contextsWithPeriodTime: list[ModelContext | None] = []
+        contextsWithPeriodTimeZone: list[ModelContext | None] = []
+        for contexts in allContexts.values():
+            for context in contexts:
+                for uncastElt in context.iterdescendants("{http://www.xbrl.org/2003/instance}startDate",
+                                                          "{http://www.xbrl.org/2003/instance}endDate",
+                                                          "{http://www.xbrl.org/2003/instance}instant"):
+                    elt = cast(Any, uncastElt)
+                    m = datetimePattern.match(elt.stringValue)
+                    if m:
+                        if m.group(1):
+                            contextsWithPeriodTime.append(context)
+                        if m.group(3):
```

Comment:
Out of curiosity, where are these `1`/`3` coming from?  From [this](https://github.com/Arelle/Arelle/blob/fb5e3de9626e8d0dfa45e6adf63369aa9df4af18/arelle/ModelValue.py#L263-L264), it looks like that would correspond to year/day components, but maybe I'm looking in the wrong place.

Is there a testcase for this?

Reply:
No test case yet. There is a conformance suite provided by the jurisdiction that we need to wire up, but havent yet.

Reply:
I had taken them from here: https://github.com/Arelle/Arelle/blob/fb5e3de9626e8d0dfa45e6adf63369aa9df4af18/arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py#L547-L552. but I had terribly messed up the imports.